// Message2Server
syntax = "proto3";
package protobuf;
import "MessageType.proto";


message ShipMsg
{
    int64 ship_id = 1;
    ShipType ship_type = 2;
}

message BuildingMsg
{
    int64 building_id = 1;
    BuildingType building_type = 2;
}
message HomeMsg
{
    PlayerTeam team = 1;
}
message CollectorMsg
{
    int64 collector_id = 1;
    CollectorType collector_type = 2;
    int64 ship_id=3;
}

message ArmorMsg
{
    int64 armor_id = 1;
    ArmorType armor_type = 2;
    int64 ship_id=3;
}

message ShieldMsg
{
    int64 shield_id = 1;
    ShieldType shield_type = 2;
    int64 ship_id=3;
}

message BuilderMsg
{
    int64 builder_id = 1;
    BuilderType builder_type = 2;
    int64 ship_id=3;
}

message BulleterMsg
{
    int64 bullet_id = 1;
    BulletType bullet_type = 2;
    int64 ship_id=3;
}

message MoveMsg
{
    int64 ship_id = 1;
    double angle = 2;
    int64 time_in_milliseconds = 3;
}



message SendMsg
{
    int64 player_id = 1;
    int64 to_player_id = 2;
    oneof message
    {
        string text_message = 3;
        bytes binary_message = 4;
    }
}

message AttackMsg // 相当于攻击
{
    int64 ship_id = 1;
    double angle = 2;
}

message IDMsg
{
    int64 ship_id = 1;
}

message RecoverMsg
{
    int64 ship_id = 1;
}

message BuildMsg
{
    int64 ship_id = 1;
    int64 building_id = 2;
    BuildingType building_type = 3;
    ShipType ship_type = 4;
}




// 基本继承于THUAI5，为了使发送的信息尽可能不被浪费，暂定不发这类大包。
// message MessageToServer
// {
//     MessageType messageType = 1;
//     int64 playerID = 2; // 消息发送者的playerID
//     PlayerType playerType = 3;
//     HumanType humanType= 4;
//     TrickerType trickerType = 5;
//     double angle = 6; // messageType为Move, Attack时的角度
//     PropType propType = 7; // messageType为PickProp时要捡起的道具类型，防止多个道具堆叠时出现问题
//     int64 timeInMilliseconds = 8;//时间参数
//     int64 ToPlayerID = 9; // messageType为Send时要发送的对象的ID
//     string message = 10; // messageType为Send时发送的消息内容
// }